custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -613,19 +655,44 @@ def run_test_predict(self):
         self.options.code_dir = self.options.output
         self.options.output = os.devnull
         __target_temp = None
-        if self.options.target:
-            __tempfile = NamedTemporaryFile()
+
+        if self.options.target or self.options.row_weights:
             df = self.input_df
-            if self.target_type == TargetType.TRANSFORM:
+            if self.target_type == TargetType.TRANSFORM and self.options.target:
                 target_df = df[self.options.target]
                 __target_temp = NamedTemporaryFile()
                 target_df.to_csv(__target_temp.name, index=False, line_terminator="\r\n")
-            df = df.drop(self.options.target, axis=1)
+
+            if self.options.target:
+                df = df.drop(self.options.target, axis=1)
+
+            if self.options.row_weights:
+                df = df.drop(self.options.row_weights, axis=1)
+
+                # If weights was included in the sparse input training data, drop it from the column file
+                # TODO: Always force weights + target to be separate files with sparse input
+                if self.options.sparse_column_file:
+                    sparse_colnames = StructuredInputReadUtils.read_sparse_column_file_as_list(
+                        self.options.sparse_column_file
+                    )
+                    if self.options.row_weights in sparse_colnames:
+                        sparse_colnames.remove(self.options.row_weights)
+
+                        with open(self.options.sparse_column_file, "w") as file:
+                            file.write("\n".join(sparse_colnames))
```

Comment:
Another case where we didn't drop row_weights from X before we performed inference :grimacing: